src/bin/pg_autoctl/keeper_pg_init.c

```diff
@@ -86,6 +88,198 @@ keeper_pg_init_fsm(Keeper *keeper, KeeperConfig *config)
 }
 
 
+/*
+ * keeper_ensure_pg_configuration_files_in_pgdata checks if
+ * postgresql.conf, pg_hba.conf, pg_ident.conf files exist
+ * in $PGDATA, if not it tries to get them from default
+ * location and modifies paths inside copied postgresql.conf.
+ */
+static bool
+keeper_ensure_pg_configuration_files_in_pgdata(KeeperConfig *config)
+{
+	char pgdata[MAXPGPATH];
+	char hbaConfPath[MAXPGPATH];
+	char posgresqlConfPath[MAXPGPATH];
+	char *pgdataRelativePath = NULL;
+	char defaultConfigurationPath[MAXPGPATH];
+	bool configurationPathPresent = false;
+	bool postgresqlConfExists = false;
+	char defaultPosgresqlConfPath[MAXPGPATH];
+	char defaultHbaConfPath[MAXPGPATH];
+	char defaultIdentConfPath[MAXPGPATH];
+	char identConfPath[MAXPGPATH];
+
+	get_absolute_path(config->pgSetup.pgdata, pgdata, MAXPGPATH);
+
+	/* check if postgresql.conf exists in pgdata. there is no separate check for pg_hba and pg_ident  */
+	join_path_components(posgresqlConfPath, pgdata, "postgresql.conf");
+	postgresqlConfExists = file_exists(posgresqlConfPath);
+	if (postgresqlConfExists)
+	{
+		return true;
+	}
+
+	/* check if pgdata is in default path. */
+	if (strstr(pgdata, "/var/lib/postgresql") != pgdata)
+	{
+		log_error("Cannot determine configuration file location for this instance. pgdata = %s", pgdata);
+		return false;
+	}
+
+	/*
+	 * We are trying to translate /var/lib/postgresql/PGVERSION/main to
+	 * /etc/postgresql/PG_VERSION/main.
+	 */
+	pgdataRelativePath = strstr(pgdata, "postgresql");
+	join_path_components(defaultConfigurationPath, "/etc", pgdataRelativePath);
+
+	/* true if we are working on debian/ubuntu installation */
+	configurationPathPresent = directory_exists(defaultConfigurationPath);
+	if (!configurationPathPresent)
+	{
+		log_error("Cannot find configuration file directory \"%s\"", defaultConfigurationPath);
+		return false;
+	}
+
+	join_path_components(defaultPosgresqlConfPath, defaultConfigurationPath, "postgresql.conf");
+	if (!move_file(defaultPosgresqlConfPath, posgresqlConfPath))
+	{
+		/* errors are already logged */
+		return false;
+	}
```

Comment:
Could this be done in a specialised function, to keep the main code here as small as possible? Otherwise we're lost in the minutia of the function here and miss the big picture.